docs/docs/Components/components-processing.md

```diff
@@ -345,107 +255,256 @@ For an additional example of using the **Parser** component to format a DataFram
+2. In the **Split Text** component, define your data splitting parameters.
+
+This example splits incoming JSON data at the separator `},{`, so each chunk contains one complete JSON object.
+
+![Split text component and chroma-db](/img/component-split-text.png)
+
+The order of precedence is **Separator**, then **Chunk Size**, and then **Chunk Overlap**.
+If any segment after separator splitting is longer than `chunk_size`, it is split again to fit within `chunk_size`.
+
+After `chunk_size`, **Chunk Overlap** is applied between the chunks to maintain context between them.
+
+3. Connect a **Chat Output** component to the **Split Text** component's **DataFrame** output to view its output.
+4. Click **Playground**, and then click **Run Flow**.
+The output contains a table of JSON objects split at `},{`.
+```text
+{
+"userId": 1,
+"id": 1,
+"title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
+"body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
+},
+```
+5. Clear the **Separator** field, and then run the flow again.
+The output contains chunked 50-character text with 10 characters of overlap.
+```text
+"title": "sunt aut facere repellat provident occae
+dent occaecati excepturi optio reprehenderit",
+erit",
+"body": "quia et suscipit\nsuscipit rec
+```
+
+### Inputs
+
+| Name | Display Name | Info |
+|------|--------------|------|
+| data_inputs | Input Documents | The data to split.The component accepts [Data](/concepts-objects#data-object) or [DataFrame](/concepts-objects#dataframe-object) objects. |
+| chunk_overlap | Chunk Overlap | The number of characters to overlap between chunks. Default: `200`. |
+| chunk_size | Chunk Size | The maximum number of characters in each chunk. Default: `1000`. |
+| separator | Separator | The character to split on. Default: `newline`. |
+| text_key | Text Key | The key to use for the text column (advanced). Default: `text`. |
+
+### Outputs
+
+| Name | Display Name | Info |
+|------|--------------|------|
+| chunks | Chunks | List of split text chunks as [Data](/concepts-objects#data-object) objects. |
+| dataframe | DataFrame | List of split text chunks as [DataFrame](/concepts-objects#dataframe-object) objects. |
+
+## Update data
+
+This component dynamically updates or appends data with specified fields.
+
+### Inputs
+
+| Name | Display Name | Info |
+|------|--------------|------|
+| old_data | Data | The records to update |
+| number_of_fields | Number of Fields | Number of fields to add (max 15) |
+| text_key | Text Key | Key for text content |
+| text_key_validator | Text Key Validator | Validates text key presence |
+
+### Outputs
+
+| Name | Display Name | Info |
+|------|--------------|------|
+| data | Data | Updated [Data](/concepts-objects#data-object) objects. |
+
+## Legacy components
```

Comment:
You should not have empty headings. Add body text to explain what's in this section.

Reply:
Added